### ISSUE TITLE
WoWUp is not updating add on providers on add on update or reinstall

### ISSUE THREAD

**Issue by MrUzagi** [WowUp/WowUp#1257]
**Describe the bug**
New or updated provider tags from the add on .TOC are not updated in the list of available providers for that add on.
An update or a re-install of the add on does not help. Only a complete uninstall of the add on and then reinstall works.

**To Reproduce**
1. Check provider list of an add on. (I'm using Rematch as an example here. Curse and Wowinterface are listed.)
2. See that the author has added a provider previously not listed for that add on. (Added Wago.)
3. Update the add on.
4. Notice that the provider list for the add on still have not changed. (Still Curse and Wowinterface.)
5. Do a re-install of the add on.
6. Notice that the list of providers still have not changed. (Still Curse and Wowinterface.)
7. Completely uninstall the add on.
8. Reinstall the add on and notice that the provider list is now updated. 

**Expected behavior**
The list of providers for the add on should be updated on update and reinstall.

**Screenshots**
Rematch v4.11.23 that according to change log and .TOC have added wago as a provider.
![image](https://user-images.githubusercontent.com/21023158/169836264-8fb8266b-756c-46c9-84e3-1be73967b05a.png)
![image](https://user-images.githubusercontent.com/21023158/169835504-4cd5c7f9-3fc2-484c-b1ff-983202746a2b.png)

After uninstall and reinstall.
![image](https://user-images.githubusercontent.com/21023158/169836691-70872c01-5b0b-4c20-a8e9-a57d7833fd65.png)

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser: I thought WoWUp was a browser. But I use FF as default elsewhere
 - Version: WoWUp v2.8.1

**Comment by linaori** [comment]
When you update and then do a re-scan, does that pick up the ID properly?

**Comment by MrUzagi** [comment]
I don't know. I try to avoid doing re-scan since I've had issues in the past, but more recently because I want to keep the links to Curse for the remaining add-ons I still haven't found another source for.

It seems probable that that could work. But at the same time I think doing a re-scan of *all* add-ons seems like overkill to fix an issue with one add-on.

**Comment by linaori** [comment]
GitHub repositories should stay known as of a recent patch

**Comment by MrUzagi** [comment]
Yes, they do. But what happens to the ones that are only on Curse atm?

**Comment by linaori** [comment]
Nothing will change, you can't access them already as is. They'll turn into "unknown" addons
